### ISSUE TITLE
`raw` always warns "raw module does not support the environment keyword"

### ISSUE THREAD

**Issue by retrospectacus** [ansible/ansible#23764]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

raw

##### ANSIBLE VERSION

```
ansible 2.4.0 (devel ffa92485b4) last updated 2017/04/19 12:14:18 (GMT -700)
  config file = 
  configured module search path = [u'/home/adama/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adama/ansible/lib/ansible
  executable location = /home/adama/ansible/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```
```
12:46 < sivel> bisected to https://github.com/ansible/ansible/commit/df5895e585e18bfbf84af993c152801124332bf0
12:46 < sivel> https://github.com/ansible/ansible/pull/23384
```

##### OS / ENVIRONMENT

Linux

##### SUMMARY

Using the `raw` command always produces this warning.
Applying this patch informs us that env is `[{}]` (empty-ish but still truthy)
```
--- a/lib/ansible/plugins/action/raw.py
+++ b/lib/ansible/plugins/action/raw.py
@@ -28,6 +28,7 @@ class ActionModule(ActionBase):
             task_vars = dict()
 
         if self._task.environment:
+            self._display.warning( self._task.environment )
             self._display.warning('raw module does not support the environment keyword')
 
         result = super(ActionModule, self).run(tmp, task_vars)
```

##### STEPS TO REPRODUCE

Any playbook or role using the raw plugin.

```yaml
- name: Simple test for raw + environment.
  hosts: localhost
  connection: local
  tasks:
    - name: Run the test.
      raw: echo hello world
```

##### EXPECTED RESULTS

No warnings to be produced

##### ACTUAL RESULTS

Warning is produced.

```
$ ansible-playbook test.yml
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [Simple test for raw + environment.] ********************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************
ok: [localhost]

TASK [Run the test.] *****************************************************************************************************************
 [WARNING]: [{}]

 [WARNING]: raw module does not support the environment keyword

changed: [localhost]

PLAY RECAP ***************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0   
```
(It does not say "WARNING [{}]" if you did not apply the debug patch above)

**Comment by sivel** [comment]
cc @bcoca

**Comment by bcoca** [comment]
possible fix https://gist.github.com/bcoca/88dcd5f1cd9202b06eeeeed909437b3c

**Comment by retrospectacus** [comment]
I did a few tests. LGTM

**Comment by retrospectacus** [comment]
test case https://gist.github.com/retrospectacus/d2e9de7bd6ab70b2a32b5687db43990b

**Comment by bcoca** [comment]
closed via 4b5553c
